packages/build/src/build/buildJs/buildJs.test.js

```diff
@@ -0,0 +1,146 @@
+  limitations under the License.
+*/
+
+import buildJs from './buildJs.js';
+import testContext from '../../test/testContext.js';
+
+test('buildJs default', async () => {
+  const context = testContext();
+  const components = {
+    pages: [{ id: 'a', style: { _js: 'return 1;' }, requests: [] }],
+  };
+
+  buildJs({ components, context });
+  expect(components.pages).toEqual([
+    { id: 'a', requests: [], style: { _js: 'tjmrJIhufQzEpj/iGu5AumDcrBQ=' } },
+  ]);
+  expect(context.jsMap).toEqual({
+    client: { 'tjmrJIhufQzEpj/iGu5AumDcrBQ=': 'return 1;' },
+    server: {},
+  });
+});
+
+test('buildJs identical function definitions', async () => {
+  const context = testContext();
+  const components = {
+    pages: [
+      { id: 'a', style: { _js: 'return 2;' }, requests: [] },
+      { id: 'b', style: { _js: 'return 10;' }, requests: [] },
+      { id: 'c', style: { _js: 'return 2;' }, requests: [] },
+    ],
+  };
+
+  buildJs({ components, context });
+
+  expect(components.pages).toEqual([
+    { id: 'a', requests: [], style: { _js: 'XjWXGQx0Ga1LvOQtdxUpR6mZwhA=' } },
+    { id: 'b', requests: [], style: { _js: 'h4uNNgee8PnSsXJuEXZQ7FSbnZY=' } },
+    { id: 'c', requests: [], style: { _js: 'XjWXGQx0Ga1LvOQtdxUpR6mZwhA=' } },
+  ]);
+
+  expect(context.jsMap).toEqual({
+    client: {
+      'h4uNNgee8PnSsXJuEXZQ7FSbnZY=': 'return 10;',
+      'XjWXGQx0Ga1LvOQtdxUpR6mZwhA=': 'return 2;',
+    },
+    server: {},
+  });
+});
+
+test('buildJs server-side functions', async () => {
+  const context = testContext();
+  const components = {
+    pages: [
+      { id: 'a', style: { _js: 'return 1;' }, requests: [{ _js: 'return 1;' }] },
+      { id: 'b', style: { _js: 'return 12;' }, requests: [{ _js: 'return 10;' }] },
+      { id: 'c', style: { _js: 'return 1;' }, requests: [{ _js: 'return 1;' }] },
+    ],
+  };
+
+  buildJs({ components, context });
+
+  expect(components.pages).toEqual([
+    {
+      id: 'a',
+      requests: [{ _js: 'tjmrJIhufQzEpj/iGu5AumDcrBQ=' }],
+      style: { _js: 'tjmrJIhufQzEpj/iGu5AumDcrBQ=' },
+    },
+    {
+      id: 'b',
+      requests: [{ _js: 'h4uNNgee8PnSsXJuEXZQ7FSbnZY=' }],
+      style: { _js: 'PGqYgLcEpG/AkAZycfKEpwxXT7Y=' },
+    },
+    {
+      id: 'c',
+      requests: [{ _js: 'tjmrJIhufQzEpj/iGu5AumDcrBQ=' }],
+      style: { _js: 'tjmrJIhufQzEpj/iGu5AumDcrBQ=' },
+    },
+  ]);
+
+  expect(context.jsMap).toEqual({
+    client: {
+      'PGqYgLcEpG/AkAZycfKEpwxXT7Y=': 'return 12;',
+      'tjmrJIhufQzEpj/iGu5AumDcrBQ=': 'return 1;',
+    },
+    server: {
+      'h4uNNgee8PnSsXJuEXZQ7FSbnZY=': 'return 10;',
+      'tjmrJIhufQzEpj/iGu5AumDcrBQ=': 'return 1;',
+    },
+  });
+});
+
+test('buildJs connection functions', async () => {
+  const context = testContext();
+  const components = {
+    pages: [
+      { id: 'a', requests: [] },
+      { id: 'b', requests: [] },
+      { id: 'c', requests: [] },
+    ],
+    connections: [
+      { id: 'c_a', style: { _js: 'return 3;' } },
```

Comment:
This isn't really valid config, connections don't have a style prop. 

It doesn't really matter now because of how the parser is implemented, but it would be more robust if the test case had valid Lowdefy config